cores/esp8266/hwdt_app_entry.cpp

```diff
@@ -0,0 +1,1051 @@
+/*
+ *   Copyright 2020 Michael Hightower
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+/*
+ * As far as I know, there is no way to get called for a Hardware WDT. I assume
+ * it generates a form of reset that occurs at a low level that cannot be
+ * trapped. Debugging an HWDT can be quite challenging.
+ *
+ * This module writes a stack dump to the serial port after a Hardware Watchdog
+ * Timer has struck, and a new boot cycle has begun. By making adjustments to the
+ * stack, we can avoid crash stack data being overwritten by this tool,
+ * the Boot ROM, and the bootloader.
+ *
+ * We are using the method defined for `core_esp8266_app_entry_noextra4k.cpp` to
+ * load an alternate `app_entry_redefinable()`. For details on this method, see
+ * comments in `core_esp8266_main.cpp's app_entry()`.
+ *
+ * Using this tool's alternate `app_entry_redefinable()`, we can gain control
+ * before the SDK is started. And dump the contents of the "sys" and "cont"
+ * stacks.
+ *
+ * By making some adjustments to start of the stack pointer, at the entry to
+ * `app_entry_redefinable()`, and also to the stack pointer passed to the SDK,
+ * we can preserve the stack during an HWDT event.
+ *
+ * To use, just place this file in the sketch directory before opening the
```

Comment:
Are these instructions still valid, given that the functionality is now integrated into the core?

Reply:
Updated.